File: mesmaeker-mockito/src/main/java/fr/xgouchet/mesmaeker/mockito/ValidatingOngoingVoidClause.java
package fr.xgouchet.mesmaeker.mockito;

import java.util.function.Consumer;

/**
 * @author Xavier F. Gouchet
 */
public class ValidatingOngoingVoidClause<T> implements OngoingVoidClause<T> {

    private final T subject;

    public ValidatingOngoingVoidClause(T subject) {
        this.subject = subject;
    }

    @Override public OngoingVoidClause<T> when(Consumer<T> consumer) {
        consumer.accept(subject);
        return this;
    }
}

File: mesmaeker-mockito/src/main/java/fr/xgouchet/mesmaeker/mockito/ValidatingOngoingVoidClause.kt
package fr.xgouchet.mesmaeker.mockito

/**
 * @author Xavier F. Gouchet
 */
class ValidatingOngoingVoidClause<T>(private val subject: T)
    : OngoingVoidClause<T> {

    override fun `when`(block: (T) -> Unit): OngoingVoidClause<T> {
        return whenever(block)
    }

    override fun whenever(block: (T) -> Unit): OngoingVoidClause<T> {
        block(subject)
        return this
    }
}
